src/material/datepicker/calendar-body.ts

```diff
@@ -122,6 +122,9 @@ export class MatCalendarBody implements OnChanges, OnDestroy {
   /** Emits when a new value is selected. */
   @Output() readonly selectedValueChange = new EventEmitter<MatCalendarUserEvent<number>>();
 
+  /** Emits when a new date becomes active. */
+  @Output() readonly activeValueChange = new EventEmitter<MatCalendarUserEvent<number>>();
```

Comment:
What is this event used for? It seems like we're only emitting it/forwarding it.

Reply:
Hmm, good question. What this is trying to accomplish is making it so that the calendar body can update the value of the `activeDate` when a cell receives the focus event. `activeDate` has a reference in three components – the root component MatCalendarBody, the component for the specific view month/year/multiyear and the calendar body.

This might be easier to do with banana in a box rather than passing an event. That is currently (in master branch) how the `MatCalendar` component gives the `activeDate` to the MatMonthView component (and also MatMonthView and MatMultiYearView).

https://github.com/angular/components/blob/03485cd63e7622da839c115679f261b15989e33c/src/material/datepicker/calendar.html#L6

Reply:
ok, so if I'm understanding it correctly, what it's trying to achieve is to sync the `activeDate` back up with the calendar if focus is moved by something we don't control (e.g. a keyboard shortcut)?

Reply:
Exactly, if I had more time, I would even go as far as renaming `activeDate` to `focusedDate`, but I'm not sure that would be worth changing public API for.